staff/acct/sorry/sorry

```diff
@@ -1,156 +1,202 @@
-ldapsearch -x uid="$sorriedUser"
-ldapsearch -x cn=sorry | tail
-finger -m "$sorriedUser"
-ls -la "$userdir"
-ls -ld "$httpdir"
-
-# logging
-echo "$(/bin/date) - $rootstaffer $sorriedUser" >> /opt/acct/sorry.log
-
-# Notify user by email if email address is available.
-email_from='Open Computing Facility <help@ocf.berkeley.edu>'
-email_subject='[OCF] Account disabled'
-email_to="$(ldapsearch "uid=${sorriedUser}" mail | grep ^mail | cut -d' ' -f2)"
-[ -n "$email_to" ] && \
-  mail -a "From: ${email_from}" -s "$email_subject" "$email_to" <<EOF
+#!/usr/bin/env python3
+# Script for sorrying OCF accounts v2
+import argparse
+import datetime
+import grp
+import json
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+import textwrap
+
+import ocflib.account.manage
+import ocflib.account.search
+import ocflib.account.utils
+import ocflib.misc.mail
+
+REASON_PATH = pathlib.Path('/opt/share/utils/staff/acct/sorry/')
+SORRY_SHELL = pathlib.Path('/opt/share/utils/bin/sorried')
+# TODO: This sorry log file does not exist anymore, do we still log to file?
+SORRY_LOG = pathlib.Path('/opt/acct/sorry.log')
+kerberos_new_ticket = True
+
+
+def find_sorry_file(file_path):
+    # Own sorry file
+    if os.path.isfile(file_path):
+        return file_path
+
+    # Default sorry files
+    default_sorry_file = os.path.join(REASON_PATH, file_path)
+    if os.path.isfile(default_sorry_file):
+        return default_sorry_file
+
+    raise FileNotFoundError('Sorry file "{}" does not exist.'.format(file_path))
+
+
+def sorry_user(username, sorry_file):
+    attributes = ocflib.account.search.user_attrs(username)
+    if attributes is None:
+        raise ValueError('User {0} not found.'.format(username))
+
+    rootstaffer = real_user()
+    kerberos_new_ticket = kerberos_init(rootstaffer)
+
+    user_dir = attributes['homeDirectory']
+
+    print('Copying sorry file and making .oldshell file')
+
+    if not os.path.isfile(user_dir + '/.oldshell'):
+        with open(user_dir + '/.oldshell', 'w') as oldshell_file:
+            oldshell = attributes['loginShell']
+            print(oldshell, file=oldshell_file)
+
+    shutil.copy2(sorry_file, user_dir + '/.sorry')
+    os.chmod(user_dir + '/.sorry', 0o400)
+    shutil.chown(user_dir + '/.sorry', user=username, group='ocf')
+
+    print("Changing user's shell to sorry shell")
+
+    ocflib.infra.ldap.modify_ldap_entry(ocflib.account.utils.dn_for_username(username),
+                                        {'loginShell': SORRY_SHELL, 'gidNumber': grp.getgrnam('sorry').gr_gid})
+
+    http_dir = ocflib.account.utils.web_dir(username)
+    print('Changing permissions on httpdir to 000')
+    if (os.path.isdir(http_dir)):
+        os.chmod(http_dir, 0o000)
+
+    print('Changing permissions on home directory to 500')
+    os.chmod(user_dir, 0o500)
+
+    # Omitting final system check for now
+    with open(sorry_file, 'r') as filep:
+        log(rootstaffer, username, filep.read())
```

Comment:
`filep.read()` can only be called once; any subsequent calls would return `''`. You probably want to save its result to a variable and use that.